spec/models/state_change_spec.rb

```diff
@@ -10,4 +10,98 @@
     it { is_expected.to respond_to :request_id }
     it { is_expected.to respond_to :action }
   end
+
+  describe "before_save calculate_new_status" do
+    subject(:state_change) { FactoryBot.create :state_change, request: request, action: action, approver: approver }
+    let(:approver) { FactoryBot.create :staff_profile }
+    context "change applied to an absence request" do
+      let(:request) { FactoryBot.create :absence_request }
+
```

Comment:
Do we have a test that the initial status is pending? I guess it doesn't go here, but it seems like we should have one somewhere.

Reply:
That is an interesting question.  I did nothing to check if the state change was valid (aka you could not go from canceled to denied). I'm thinking the workflow would handle the more complex issues around approval, but maybe that should be handled here.

Reply:
That makes sense, that we can do this as part of the state machine work.

Reply:
Although, if no one ever manually sets a status maybe we don't need a state machine?

Reply:
We need to do a bunch of things when the state changes, like multiple notifications.  I do not think I want that in a before save, but maybe not a state machine.

Reply:
yeah it seems like we may still want it. we can talk about it more when the time comes!